### PR TITLE
subsys: event_manager: Shell integration

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds shell integration to Event Manager.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@pdunaj, @Qbicz, could you review ?

File: subsys/event_manager/event_manager_shell.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+			evt_mask |= BIT(ev_id);
+		}
+
+		shell_fprintf(shell,
+			      SHELL_NORMAL,
+			      "Displaying all events %sabled\n",
+			      enable ? "en":"dis");
+	} else {
+		const size_t index_cnt = argc - 1;
+		int event_indexes[index_cnt];
```
Comment (pdunaj) [inline subsys/event_manager/event_manager_shell.c]: I would prefer `event_index[argc - 1]` 
i.e. not using tmp variable

File: subsys/event_manager/event_manager_shell.c [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+		for (size_t i = 0; i < index_cnt; i++) {
+			char *end;
+
+			event_indexes[i] = strtol(argv[i + 1], &end, 10);
+
+			if (event_indexes[i] < 0 ||
```
Comment (pdunaj) [inline subsys/event_manager/event_manager_shell.c]: `|| (end != NULL)`

File: subsys/event_manager/event_manager_shell.c [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+		      sizeof(event_manager_displayed_events)*8 - 1),
+	SHELL_CMD_ARG(enable, NULL, "Enable displaying event with given ID",
+		      enable_event_displaying, 0,
+		      sizeof(event_manager_displayed_events)*8 - 1),
+	SHELL_SUBCMD_SET_END
+};
+
+SHELL_CMD_REGISTER(event_manager, &sub_event_manager,
+		"Event Manager commands", NULL);
+
```
Comment (pdunaj) [inline subsys/event_manager/event_manager_shell.c]: excessive empty line

File: subsys/event_manager/event_manager.c [hunk 4]
```diff
@@ -10,6 +10,13 @@
 #include <event_manager.h>
 #include <logging/log.h>
 
+#ifdef CONFIG_SHELL
+extern u32_t event_manager_displayed_events;
+#else
+/* By default, when there is no shell, all events are shown */
+u32_t event_manager_displayed_events = 0xffffffff;
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply subsys/event_manager/event_manager.c]: done

File: subsys/event_manager/event_manager.c [hunk 5]
```diff
@@ -60,7 +67,10 @@ static void event_processor_fn(struct k_work *work)
 		const struct event_type *et = eh->type_id;
 
 		trace_event_execution(eh, true);
-		if (IS_ENABLED(CONFIG_DESKTOP_EVENT_MANAGER_SHOW_EVENTS)) {
+		if (IS_ENABLED(CONFIG_DESKTOP_EVENT_MANAGER_SHOW_EVENTS)
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply subsys/event_manager/event_manager.c]: done

File: subsys/event_manager/event_manager_shell.c [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <event_manager.h>
+
+u32_t event_manager_displayed_events;
+
+static int show_events(const struct shell *shell, size_t argc,
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

File: subsys/event_manager/event_manager_shell.c [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+static void set_event_displaying(const struct shell *shell, size_t argc,
+				 char **argv, bool enable)
+{
+	u32_t evt_mask = 0;
+
+	/* If no IDs specified, all registered events are affected */
+	if (argc == 1) {
+		for (const struct event_type *et = __start_event_types;
+		(et != NULL) && (et != __stop_event_types); et++) {
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

File: subsys/event_manager/event_manager_shell.c (hunk 3 above)
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

File: subsys/event_manager/event_manager_shell.c [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+			evt_mask |= BIT(ev_id);
+		}
+
+		shell_fprintf(shell,
+			      SHELL_NORMAL,
+			      "Displaying all events %sabled\n",
+			      enable ? "en":"dis");
+	} else {
+		const size_t index_cnt = argc - 1;
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

File: subsys/event_manager/event_manager_shell.c [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+		int event_indexes[index_cnt];
+
+		for (size_t i = 0; i < index_cnt; i++) {
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

File: subsys/event_manager/event_manager_shell.c (hunk 1 above)
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

File: subsys/event_manager/event_manager_shell.c (hunk 2 above)
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

**Comment by MarekPieta** [comment]
I made requested changes

**Comment by pdunaj** [comment]
Hi @kapi-no , could you review?

File: subsys/event_manager/event_manager_shell.c [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+		int event_indexes[argc - 1];
+
+		for (size_t i = 0; i < ARRAY_SIZE(event_indexes); i++) {
+			char *end;
+
+			event_indexes[i] = strtol(argv[i + 1], &end, 10);
+
+			if (event_indexes[i] < 0
+			    || event_indexes[i] >= __stop_event_types
+				- __start_event_types
```
Comment (KAGA164) [inline subsys/event_manager/event_manager_shell.c]: It is a bit unreadable to me. Please move this to previous line.

File: subsys/event_manager/event_manager_shell.c [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+
+		__ASSERT_NO_MSG(el != NULL);
+		shell_fprintf(shell, SHELL_NORMAL, "|\t[L:%s]\n", el->name);
+	}
+
+	return 0;
+}
+
+static int show_subscribers(const struct shell *shell, size_t argc,
+		char **argv)
```
Comment (KAGA164) [inline subsys/event_manager/event_manager_shell.c]: Alignment

File: subsys/event_manager/event_manager_shell.c [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+				'E' : 'D',
+			      ev_id,
+			      et->name);
+	}
+
+	return 0;
+}
+
+static int show_listeners(const struct shell *shell, size_t argc,
+		char **argv)
```
Comment (KAGA164) [inline subsys/event_manager/event_manager_shell.c]: Alignment

File: subsys/event_manager/event_manager_shell.c (hunk 10 above)
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: ok, done

File: subsys/event_manager/event_manager_shell.c (hunk 11 above)
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

File: subsys/event_manager/event_manager_shell.c (hunk 12 above)
Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

**Comment by MarekPieta** [comment]
Fixed

**Comment by MarekPieta** [comment]
Can we merge ?

**Comment by pdunaj** [comment]
Hi @KAGA164 , could you re-review?

File: subsys/event_manager/event_manager_shell.c [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+			      enable ? "en":"dis");
+	} else {
+		int event_indexes[argc - 1];
+
+		for (size_t i = 0; i < ARRAY_SIZE(event_indexes); i++) {
+			char *end;
+
+			event_indexes[i] = strtol(argv[i + 1], &end, 10);
+
+			if (event_indexes[i] < 0
```
Comment (kapi-no) [inline subsys/event_manager/event_manager_shell.c]: You can put each condition in brackets

Comment (MarekPieta) [reply subsys/event_manager/event_manager_shell.c]: done

**Pull request merged**